src/api/app/models/bs_request.rb

```diff
@@ -62,6 +64,17 @@ def save!
     notify if new
   end
 
+  def self.find_by_number!(number)
```

Comment:
I don't like this at all. A normal `.find_by!` would already raise `ActiveRecord::RecordNotFound`....